lib.php

```diff
@@ -0,0 +1,286 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Step subplugin to notify students of a course that the course is being deleted.
+ *
+ * @package    lifecyclestep_notifystudents
+ * @copyright  2021 Aaron Koßler WWU
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace tool_lifecycle\step;
+
+use context_course;
+use core_user;
+use tool_lifecycle\local\manager\process_manager;
+use tool_lifecycle\local\manager\settings_manager;
+use tool_lifecycle\local\manager\step_manager;
+use tool_lifecycle\local\response\step_response;
+use tool_lifecycle\settings_type;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/../lib.php');
+
+/**
+ * Step subplugin to notify students of a course that the course is being deleted.
+ *
+ * @package    lifecyclestep_notifystudents
+ * @copyright  2021 Aaron Koßler WWU
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class notifystudents extends libbase {
+
+    /**
+     * Processes the course and returns a response.
+     * The response tells either
+     *  - that the subplugin is finished processing.
+     *  - that the subplugin is not yet finished processing.
+     *  - that a rollback for this course is necessary.
+     * @param int $processid of the respective process.
+     * @param int $instanceid of the step instance.
+     * @param mixed $course to be processed.
+     * @return step_response
+     * @throws \coding_exception
+     * @throws \dml_exception
+     */
+    public function process_course($processid, $instanceid, $course) {
+        // TODO Write db function insert_db($type).
```

Comment:
Here is still a todo. I'm not completely sure, what it means, in my opinion you don't necessarily have to refactor anything, but feel free to do so, if you want to :)

Reply:
I think because it is pretty much the same code if I can get the enrolled students with a capability it will be easier to create a function from it.